### PR TITLE
feat: parse triple backtick strings, discarding the info header

### PR DESCRIPTION
**Pull request by sxlijin**
This is a more LLM-reliable way to get codegen outputs

We should add a documentation guide page for this (possibly a great community contribution candidate here), but in the meantime I've drafted up a blog post

----

> [!IMPORTANT]
> Add support for parsing triple backtick strings in JSON parser, including dedenting and discarding info headers.
> 
>   - **Behavior**:
>     - Add `TripleBacktickString` variant to `JsonCollection` in `json_collection.rs` to handle triple backtick strings.
>     - Update `process_token()` in `json_parse_state.rs` to parse triple backtick strings, discarding the info header and dedenting content.
>   - **Utilities**:
>     - Move `dedent` function to `dedent.rs` in `bstd` and update its usage in `expression.rs` and `json_collection.rs`.
>   - **Tests**:
>     - Add tests in `test_code.rs` to verify parsing of triple backtick strings, including edge cases like nested backticks and dedenting.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=BoundaryML%2Fbaml&utm_source=github&utm_medium=referral)<sup> for 3d5cd23f9dcedaaa3e8b5b96cebb25112edbc857. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: engine/baml-lib/jsonish/src/tests/test_code.rs [hunk 1]
```diff
@@ -357,3 +407,220 @@ export default query(async (ctx) => {
...
+```",
+    "type": "code",
+}
+```
+
+  "#,
+  FieldType::class("Test"),
+  {
+    "type": "code",
+    "code": "```\nconst { execSync } = require('child_process');\n```",
```
Comment (hellovai) [inline engine/baml-lib/jsonish/src/tests/test_code.rs:620]: i thought we said to trim + dedent?

**Review by hellovai — COMMENTED** [review]
dedent? everything else looks good!!

File: engine/baml-lib/jsonish/src/tests/test_code.rs (hunk 1 above)
Comment (sxlijin) [reply engine/baml-lib/jsonish/src/tests/test_code.rs:620]: This one is triple backticks inside a string, that's why the newlines get preserved!
